### PR TITLE
Add the Hydrogen Knight Rigsuit

### PR DESCRIPTION
**Pull request by r4d6**
Add the Hydrogen Knight rig suit, an hydrogen-powered beast of an armored RIG which is Soteria's answer to the Church's Crusader/Fortress Armor. It is expensive and long as fuck to make and only a skilled Roboticist or CRO are trained to fabricate it.

Due to the materials, components & time used in its construction, alongside its power source, it has no slowdown when active, a huge armor rating and is EMP-proof.

It is powered by an hydrogen fuel cell via a converter that can turn any fuel cell into a 30 000 rating non-rechargable cell.

Here are the steps needed to craft it : 
1. Add any RIG as a base
2. Any clothing item.
3. Any belt.
4. Soteria's MK2 Armor.
5. Soteria's MK2 Helmet
6. Phasic Scanning Module
7. Robot Camera
8. Diagnosis Unit
9. Photonic Processor Board, the small version work too.
10. 10 Cables
11. 20 Cutting Quality
12. 60 Steel Sheets
13. 45 Welding
14. 45 Drilling
15. 45 Cutting
16. 5 Plasmaglass Sheets.
17. 45 Screwdriving
18. 30 Cables
19. 30 Plasteel Sheets
20. High-Power Motor.
21. Super Capacitor
22. Booster toolmod
23. Super Matter Bin
24. Hydrogen Fuel Cell
25. Pico-Manipulator
26. Thunder Shunt
27. Rocket Engine
28. Guard toolmod
29. Omega-cell
30. 10 Pulsing
31. VTEC Borg upgrade
32. Durand Head
33. Durand Torso
34. Ripley Left Arm
35. Ripley Right Arm
36. Gygax Left Leg
37. Gygax Right Leg
38. Magboots.
39. 5 Osmium Sheets
40. Hydraulic Toolmod
41. Ion Jaunt component, specially researched in the Hydrogen weaponry node. (Cost 5 Steel, 5 Plasteel, 1 Plasmaglass, 1 Diamond, 1 Tritium, 1 Osmium, 1 Metallic Hydrogen, & 20 of each **chemical** : Iron, Uranium, & Plasma)
42. 30 Hammering
43. 60 Welding.

In exchange for being the item with the longest & costliest crafting queue in the whole codebase, it has the following armor values : 80 Melee, 80 Bullet, 80 Energy, 70 Bomb, 100 Bio, 100 Rad.
And you give up being able to enter exosuits because you are too bulky.

Sprites were started by Churro and finished/reworked by HOUND.

Putting this up as a Draft PR so that people can check the code and give their opinion while waiting for HOUND to finish the sprites.

### CODE REVIEW

File: code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/obj/item/clothing/suit/space/rig/hydrogen_knight
+	name = "hydrogen knight suit"
+	armor_list = list(melee = 65, bullet = 65, energy = 65, bomb = 70, bio = 100, rad = 100) // Same stats as the crusader armor
+
+/obj/item/clothing/head/helmet/space/rig/hydrogen_knight
+	name = "hydrogen knight helm"
+	armor_list = list(melee = 65, bullet = 65, energy = 65, bomb = 70, bio = 100, rad = 100) // Same stats as the crusader helmet
+
+/obj/item/clothing/shoes/magboots/rig/hydrogen_knight
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:35]: Shoes and gloves sould also get armored imo

Comment (Dongels) [reply code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:35]: a good basis would be go off what the combat boots/gloves have.

Comment (r4d6) [reply code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:35]: The combat gauntlets & boots, or at least the rig version, has no armor value either.

Comment (Trilbyspaceclone) [reply code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:35]: I think they use the base rig armor by default could be wrong tho

**Comment by Kazkin** [comment]
Draft

**Comment by r4d6** [comment]
> Draft

Still a draft because I am waiting on HOUND to finish the sprites.

File: code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		return
+
+	toggle_piece("helmet",wearer)
+
+// This shit is hardcoded to use the parent proc of toggle_helmet, so we need to override it here.
+/obj/item/rig/hydrogen_knight/Initialize()
+	..()
+	if(helm_type)
+		verbs -= /obj/item/rig/proc/toggle_helmet
+		verbs += /obj/item/rig/hydrogen_knight/toggle_helmet
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:52]: This is a item path not a proc or verb

File: code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	suitable_cell = /obj/item/cell/large
+	initial_modules = list(/obj/item/rig_module/storage/expanded)
+	extra_allowed = list(
+		/obj/item/hydrogen_fuel_cell
+	)
+
+	req_access = list()
+	req_one_access = list()
+
+/obj/item/rig/hydrogen_knight/toggle_helmet()
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:28]:
```suggestion
/obj/item/rig/hydrogen_knight/verb/toggle_helmet()
```

File: code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm (hunk 2 above)
Comment (Trilbyspaceclone) [inline code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:52]:
```suggestion
		verbs += /obj/item/rig/hydrogen_knight/verb/toggle_helmet()
```

**Review by Trilbyspaceclone — CHANGES_REQUESTED** [review]
My only issue is the large hydrogen cell. I can look past everything else

File: code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm (hunk 3 above)
Comment (r4d6) [reply code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:28]: Tried this, did not compile

File: code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm (hunk 2 above)
Comment (r4d6) [reply code/modules/clothing/spacesuits/rig/suits/hydrogen_knight.dm:52]: Tried this, did not compile

**Comment by cdb-is-not-good** [comment]
armor_list = list(melee = 80, bullet = 80, energy = 80, bomb = 70, bio = 100, rad = 100)
My issue with this PR remains the fact that the armor values are completely bonkers. Even for something as hard as this is to get - they are just /mega/ high.

**Comment by r4d6** [comment]
> armor_list = list(melee = 80, bullet = 80, energy = 80, bomb = 70, bio = 100, rad = 100) My issue with this PR remains the fact that the armor values are completely bonkers. Even for something as hard as this is to get - they are just /mega/ high.

That's a fair point, but if the values were the same as, say, the Juggernaut Armor, AKA 60 in melee, bullet & energy, then there wouldn't any reason to choose the Knight suit over the Juggernaut suit when the second one is easier to make and give the same amount of protection.

**Comment by cdb-is-not-good** [comment]
> > armor_list = list(melee = 80, bullet = 80, energy = 80, bomb = 70, bio = 100, rad = 100) My issue with this PR remains the fact that the armor values are completely bonkers. Even for something as hard as this is to get - they are just /mega/ high.
> 
> That's a fair point, but if the values were the same as, say, the Juggernaut Armor, AKA 60 in melee, bullet & energy, then there wouldn't any reason to choose the Knight suit over the Juggernaut suit when the second one is easier to make and give the same amount of protection.

The juggernaught suit is intended to be the best suit of armor in the game, bar none - it requires extensive interdepartment cooperation, not just "give me some boosters"(which can also be found outside of the guild.). This feels like powercreep and armor powercreep is something we should be especially weary of seeing as it can make combat a slog - or allow a side to utterly stomp. And there's one other thing.

>emp_protection = 100 // Osmium plating + Ion Jaunter mean it is pretty much immune to EMPs
Fuck this, fuck this, fuck this. Fuck no.

**Comment by r4d6** [comment]
> The juggernaught suit is intended to be the best suit of armor in the game, bar none - it requires extensive interdepartment cooperation, not just "give me some boosters"(which can also be found outside of the guild.). This feels like powercreep and armor powercreep is something we should be especially weary of seeing as it can make combat a slog - or allow a side to utterly stomp. And there's one other thing.

The Juggernaut suit [require](https://github.com/sojourn-13/sojourn-station/blob/master/code/datums/craft/recipes/repairs.dm#L220) as a whole : 
- 2 Melee armor mods.
- 2 Ranged armor mods.
- 2 Rubbermesh.

All of which can be acquired from the guild, and the last one can be found. So the 'extensive interdepartment cooperation' is just "Give me those 3-6 things" to a guild adept, which is on the same level as the 'give me some boosters' you mentioned.
So, while the Hydrogen Knight suit require 2 less item from the guild than the Juggernaut suit, it compensate by being way longer to make and requiring high-end research.
Also the hydrogen knight rig will take longer to make than the juggernaut suit.

> Fuck this, fuck this, fuck this. Fuck no. It was said before when you first added ion-resistance like this that it'd get crept in and it's an awful balance change that nullifies one of the few weaknesses your suit would otherwise have.

1. The ion resistance augment wasn't merged.
2. The ion resistance augment was limited to prosthetic limbs.
3. It wouldn't be 'the best of the best' that Soteria could offer if it could be destroyed by a random person with an ion gun. And removing the ion protection would also mean removing the osmium requirement & the ion jaunter, which is the single most expensive piece of tech used in the suit's recipe, thus making the suit easier to make.

**Comment by cdb-is-not-good** [comment]
> * 2 Melee armor mods.
> * 2 Ranged armor mods.
> * 2 Rubbermesh.
> 
> All of which can be acquired from the guild, and the last one can be found. So the 'extensive interdepartment cooperation' is just "Give me those 3-6 things" to a guild adept, which is on the same level as the 'give me some boosters' you mentioned. So, while the Hydrogen Knight suit require 2 less item from the guild than the Juggernaut suit, it compensate by being way longer to make and requiring high-end research. Also the hydrogen knight rig will take longer to make than the juggernaut suit.

Your suit requires, essentially - only actual materials from Lonestar and research/produce from Soteria itself. Whereas the Juggernaught suit at least requires some /modicum/ of cooperation between two departments. You can talk up the time/material investment all you want - but at the end of the day it doesn't require anything that the average miner can't provide on a completely normal shift. there is nothing that stops this suit from being developed any round that there are miners present. 

> 
> 1. The ion resistance augment wasn't merged.
> 2. The ion resistance augment was limited to the prosthetic limb it was installed on.
> 3. It wouldn't be 'the best of the best' that Soteria could offer if it could be destroyed by a random person with an ion gun. And removing the ion protection would also mean removing the osmium requirement & the ion jaunter, which is the single most expensive piece of tech used in the suit's recipe, thus making the suit easier to make.

The ion resistance IS present in your suit though - there is not a single, actual downside to your armor outside of its time/material investment. Why is it ion resistant? If EMP immunity is only a bit of osmium away why is it not more common? This doesn't at all fit with existing precedent about how these things work, nor does it offer meaningful alternative. This is "I don't like EMPs, so i am nullifying them in full." and that is awful. Even exos still have EMP weakness!

**Comment by r4d6** [comment]
> The ion resistance IS present in your suit though - there is not a single, actual downside to your armor outside of its time/material investment. Why is it ion resistant? If EMP immunity is only a bit of osmium away why is it not more common? This doesn't at all fit with existing precedent about how these things work, nor does it offer meaningful alternative.

Up until recently, **_A L L_** of the greyson mobs were immune to EMPs, seemingly from their osmium plating.
Also, it isn't ion resistant **only** because it has osmium plating, it also has the Ion Jaunter, who's crafting require the use of a Stasis Beaker in other to not make a chemical EMP.




> Why not reduce the stats and reduce the investment? Why not make it strong in more interesting ways than just "I have bigger numbers now"?

Well, first there isn't a lot of ways to make armor strong in more interesting ways than 'I have bigger numbers', because all of the armors rely on having bigger numbers. The Juggernaut Armor is considered the "best" armor **because** it has the bigger number (And even then the Crusader Armor has bigger numbers than that).

Reducing the stats and reducing the investment will just make it an objectively worse armor, especially as it is intended to be the best that Soteria can produce. There would be no need for Soteria to make Durands if the Ivan had has much protection for way less material cost.

The investment is meant to _justify_ the stats the armor get, however strong or weak it is. However, if two armor have the same stats, then it come down to which one has the bigger investment.
If I make the hydrogen knight rig have the exact same stats and the exact same investment as the Juggernaut Suit, then I might as well just **reskin** the damn suit instead! It would be less effort and I would end up with the exact same result.

Sure, the suit is strong as fuck, and that's intended. It is also intended to take long as fuck and a fuck-ton of materials to make. Science might just require one miner doing its job to get enough mats, but all the materials that are going into the suit will only benefit one player. You could probably arm a dozen players with high-end weaponry with what is going into the suit.

Also, not only can the guild player just... start the assembly and let someone else finish it, [turns out you can LOOT the armor mods too!](https://github.com/sojourn-13/sojourn-station/blob/master/code/game/objects/random/guns.dm#L317).

In any case, any suit that Soteria can make that is the equivalent to the crusader armor will also be the equivalent of the juggernaut armor, and if that's the case, then no matter what I do, it will either be way easier to craft it than the juggernaut armor, or way harder.

**Comment by cdb-is-not-good** [comment]
>Up until recently, A L L of the greyson mobs were immune to EMPs, seemingly from their osmium plating.
Also, it isn't ion resistant only because it has osmium plating, it also has the Ion Jaunter, who's crafting require the use of a Stasis Beaker in other to not make a chemical EMP.

Mobs, dumb animals which are not able to be nearly as dangerous as players. The Jaunter still ONLY requires time and materials. How about you make it require something from other departments? Require /some/ degree of cooperation. 

>Well, first there isn't a lot of ways to make armor strong in more interesting ways than 'I have bigger numbers', because all of the armors rely on having bigger numbers. The Juggernaut Armor is considered the "best" armor because it has the bigger number (And even then the Crusader Armor has bigger numbers than that). If you could tell me how I can make an armor strong in an """interesting""" way, I'm all hear.

It's not interesting, you're right - it's just a piece of heavy armor. But what do you know? it has DOWNSIDES! It's HEAVY when you make it, it has slowdown(which your suit also doesn't.).  How do you make it interesting? Flynn suggested what, making damage draw directly from the cell. That's an option that justifies the insane defensive values.

>The investment is meant to justify the stats the armor get, however strong or weak it is.

it doesn't, not even close.  This very school of thought is an issue! There are quantities of power that cannot be justified, especially not for what you don't seem to realize is not nearly as high as it would need to be to justify the sheer utility. The best suit of armor in the ENTIRE GAME. Available to Soteria ANY round they have a miner and a single researcher to blaze through the tree. 

>Sure, the suit is strong as fuck, and that's intended. It is also intended to take long as fuck and a fuck-ton of materials to make.

It's TOO good is the problem, it's completely out of bounds with our game balance. 

>[turns out you can LOOT the armor mods too!](https://github.com/sojourn-13/sojourn-station/blob/master/code/game/objects/random/guns.dm#L317).

in ARMOR AND DUNGEON LOOT TABLES! 

>In any case, any suit that Soteria can make that is the equivalent to the crusader armor will also be the equivalent of the juggernaut armor, and if that's the case, then no matter what I do, it will either be way easier to craft it than the juggernaut armor, or way harder.

It's not equivalent to the crusader armor, is the issue. it's SIGNIFICANTLY better, it's better than crusader - better than juggernaught. There is no armor that stands up to the sheer stats, the sheer lack of downsides this suit has! A rig that can load any number of different modules, full EMP immunity and the best damage stats bar none - don't forget being able to turn your fuel-cells into THIRTY KAY cells!

**Comment by r4d6** [comment]
> It's not equivalent to the crusader armor, is the issue. it's SIGNIFICANTLY better, it's better than crusader - better than juggernaught. There is no armor that stands up to the sheer stats, the sheer lack of downsides this suit has! A rig that can load any number of different modules, full EMP immunity and the best damage stats bar none - don't forget being able to turn your fuel-cells into THIRTY KAY cells!

And it is why it has the longest crafting recipe of the entire game, requiring multiple items that are also costly to make & research.

Yes, it is the strongest armor in the whole game. However it is my hope that the long & complicated crafting recipe (Only available to the CRO & Roboticist) discourage people for rushing it uniquely to be able to unga.
Now, if people want to unga, I can't do shit to stop them. What I can do is have the armor be a hassle to make so that they'll settle for the weaker Mk2 armor instead.

If we do start seeing the armor every shift, **then** I might change my mind. But considering that I rarely see the hydrogen weaponry, even when it deal 100+ damage per shot, and is way cheaper than the suit to make, I highly doubt that people will rush for the suit all shift, every shift.

**Comment by r4d6** [comment]
Worse thing worse, I can just comment out the crafting recipe and keep it admin-only, that way Soteria would only be able to whip it out for special events.

**Comment by r4d6** [comment]
All the sprites are in, so this is now ready to be merged.

**Comment by ThatGuyFromThatPlace** [comment]
Absurd power cell, absurd armour, no speed penalty.
Boring and broken, no interesting gimmick

Also, to back up CDB's point, no amount of crafting requirements is going to counter the fact that this triumphs all other armour pieces in the ENTIRE GAME and, the fact that it exists. 

Just by existing, it is contributing HUGELY to the powercreep and powergaming encouragement that this **roleplay** server has.

It is simply the best armour with less downsides than other armour. How is that a good addition? How is that interesting? How is that not just more power creep?

An interesting idea that I keep bringing up that would add a fun gimmick to this armour is, make that absurd power cell not just a massive bonus this armour has for some arbitrary reason. Make it so that the suit's absurd defenses are powered by said power cell, draining the power cell's charge when the armour blocks damage. If the cell runs out, you lose some of the defense from the suit (Still some left, it's still a chunky suit of armour after all)
And for the love of god, have _some_ slowdown by default when powered.

**Pull request merged**